File: components/StatusDot.js
import { View, Text, StyleSheet } from 'react-native';

export default function StatusDot({ connected }) {
  return (
    <View style={styles.row}>
      <View style={[styles.dot, connected ? styles.on : styles.off]} />
      <Text style={styles.label}>{connected ? 'Band connected' : 'Searching...'}</Text>
    </View>
  );
}

const styles = StyleSheet.create({
  row: { flexDirection: 'row', alignItems: 'center', gap: 8 },
  dot: { width: 12, height: 12, borderRadius: 6 },
  on: { backgroundColor: '#1D9E75' },
  off: { backgroundColor: '#999' },
  label: { fontSize: 14, color: '#555' },
});

File: safelink_flutter/lib/widgets/status_dot.dart
import 'package:flutter/material.dart';

/// Small coloured dot with a text label indicating BLE connection state.
/// Green (#1D9E75) when connected, grey when searching/disconnected.
class StatusDot extends StatelessWidget {
  final bool connected;

  const StatusDot({super.key, required this.connected});

  @override
  Widget build(BuildContext context) {
    final color = connected
        ? const Color(0xFF1D9E75) // teal/success
        : const Color(0xFF9E9E9E); // grey

    final label = connected ? 'Band connected' : 'Searching for band...';

    return Row(
      mainAxisSize: MainAxisSize.min,
      children: [
        Container(
          width: 10,
          height: 10,
          decoration: BoxDecoration(
            color: color,
            shape: BoxShape.circle,
            boxShadow: connected
                ? [
                    BoxShadow(
                      color: color.withAlpha(120),
                      blurRadius: 6,
                      spreadRadius: 1,
                    ),
                  ]
                : null,
          ),
        ),
        const SizedBox(width: 8),
        Text(
          label,
          style: TextStyle(
            fontSize: 13,
            fontWeight: FontWeight.w500,
            color: color,
          ),
        ),
      ],
    );
  }
}
